### PR TITLE
W-11798994: Provide an isolated messaged counter that resets on every query

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

**Comment by elrodro83** [comment]
--validate

File: core/src/main/java/org/mule/runtime/core/api/management/stats/FlowConstructStatistics.java [hunk 1]
```diff
@@ -100,4 +100,45 @@ default void incConnectionErrors() {}
...
+
+  /**
+   * Provides a counter for {@link #getExecutionErrors() execution errors} that is not affected by calls to {@link #clear()} or
+   * {@link ResetOnQueryCounter#getAndReset()} calls to other instances of {@link ResetOnQueryCounter}.
+   * 
+   * @return a counter for {@link #getExecutionErrors()}.
+   * 
+   * @since 4.5
+   */
+  ResetOnQueryCounter getExecutionErrorsCounter();
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/api/management/stats/FlowConstructStatistics.java:112]: as a consumer, I would expect this counter to be initialized with the real counter's current value, or at the very least, have the option to.

Looking at `DefaultResetOnQueryCounter` it seems like this counter always starts at zero, which generates a race-condition between the agent and the first processed messages

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mule/runtime/core/internal/construct/AbstractFlowConstruct.java [hunk 2]
```diff
@@ -65,6 +65,8 @@
  */
 public abstract class AbstractFlowConstruct extends AbstractExecutableComponent implements FlowConstruct, Lifecycle {
 
+  public static final String FLOW_FLOW_CONSTRUCT_TYPE = "Flow";
```
[the remark this answers is not in the document]
Comment (elrodro83) [reply core/src/main/java/org/mule/runtime/core/internal/construct/AbstractFlowConstruct.java:68]: no, it is actually the flow_construct type for flows

File: core/src/main/java/org/mule/runtime/core/api/management/stats/FlowConstructStatistics.java [hunk 3]
```diff
@@ -100,4 +100,45 @@ default void incConnectionErrors() {}
   default long getConnectionErrors() {
     return 0;
   }
+
+  /**
+   * Provides a counter for {@link #getExecutionErrors() execution errors} that is not affected by calls to {@link #clear()} or
```
[the remark this answers is not in the document]
Comment (elrodro83) [reply core/src/main/java/org/mule/runtime/core/api/management/stats/FlowConstructStatistics.java:105]: that was the requirement, a differnet client code may call clear and we don't want that to affect the statitics collection for metering

**Review by marianogonzalez — DISMISSED** [review]

**Review by marianogonzalez — APPROVED** [review]

**Review by marianogonzalez — APPROVED** [review]

**Pull request merged**